src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp

```diff
@@ -166,6 +166,56 @@ struct convolution_impl : typed_primitive_impl_ocl<convolution> {
             conv_params.quantization = kernel_selector::QuantizationType::NONE;
         }
 
+        auto can_swap_xy = [&](kernel_selector::convolution_params& cp) -> bool {
+            if (cp.inputs[0].Batch().v == 1
+                && cp.inputs[0].GetLayout() == kernel_selector::Tensor::DataLayout::bfyx
+                && cp.inputs[0].X().v == 1 && cp.inputs[0].Y().v > 1
+                && cp.inputs[0].X().pad.Total() == 0
+                && cp.outputs[0].GetLayout() == kernel_selector::Tensor::DataLayout::bfyx
+                && cp.outputs[0].X().v == 1 && cp.outputs[0].Y().v > 1
+                && cp.weights.X().v == 1 && cp.weights.Y().v > 1
+                && !((cp.groups == cp.inputs[0].Feature().v && cp.inputs[0].Feature().v == cp.outputs[0].Feature().v && cp.split == 1)
+                    || (cp.split == cp.inputs[0].Feature().v && cp.groups == 1)) // except depth-wise conv
```

Comment:
what about being more clear on comment: `don't swap if it is depthwise conv`

Reply:
Added